java/java.source.base/src/org/netbeans/api/java/source/SourceUtils.java

```diff
@@ -683,7 +683,7 @@ private abstract static class Match {
         }
 
         final boolean apply(final FileObject fo) {
-            if (name.equals(fo.getNameExt())) {
+            if (fo.getNameExt().equals(name)) {
```

Comment:
good that getNameExt() returns empty Strings. This makes it better than the JDK 20 proposal of the Path API which was luckily dropped last minute because of it returning null:
https://bugs.openjdk.org/browse/JDK-8297814

I already had a [code inspection](https://github.com/mbien/jackpot-inspections/commit/89c3f68840351b745d16f9cc4a0b4191e6afe1d6) for that ready expecting it to be a common bug.

new version is the PR 11545 which might make it into JDK 21 following the same null safe logic

good choice back then @jtulach!